src/core/surface/init.c

```diff
@@ -73,3 +73,13 @@ void grpc_shutdown(void) {
   }
   gpr_mu_unlock(&g_init_mu);
 }
+
+int grpc_is_initialized(void) {
+  int r;
+  gpr_once_init(&g_init, do_init);
```

Comment:
I got slightly confused btw - I think "do_init" might be a misnomer. But I don't see any better name for it.